### PR TITLE
Adjust options for setup function

### PR DESCRIPTION
**Pull request by peterfication**
This way, it's clear to the user how to use the plugin by just looking at the example setup scripts and the presentend default options.

It also allows for other possible options in the future.

This also adds tests for the options.

This PR partially addresses the problems of https://github.com/johmsalas/text-case.nvim/issues/45

### CODE REVIEW

File: lua/textcase/plugin/presets.lua
Comment (peterfication) [inline lua/textcase/plugin/presets.lua:1]: The diff of this file is a bit unfortunate because it just mostly moves the register calls into a local function and I thought it makes sense to move the local functions at the top.

File: tests/textcase/plugin/options/default_keymappings_false_spec.lua [hunk 1]
```diff
@@ -0,0 +1,39 @@
+local textcase = require("textcase")
+
+local function get_buf_lines()
```
Comment (peterfication) [inline tests/textcase/plugin/options/default_keymappings_false_spec.lua:3]: TODO if this PR is merged: Create test helpers to share them across different files.

**Comment by peterfication** [comment]
@johmsalas please let me know what you think about this change.

File: lua/textcase/plugin/presets.lua [hunk 2]
```diff
@@ -100,4 +64,51 @@ M.setup = function(opts)
...
+    api.to_lower_phrase_case,
+  })
+end
+
+M.setup = function(opts)
+  local prefix = opts and opts.prefix or "ga"
+
+  local default_keymappings = true
+  if opts and opts.default_keymappings ~= nil then
+    default_keymappings = opts.default_keymappings
```
Comment (peterfication) [inline lua/textcase/plugin/presets.lua:106]: Maybe rename it to `default_keymappings_enabled` to make it clear that it's a boolean value.

**Comment by peterfication** [comment]
Needs https://github.com/johmsalas/text-case.nvim/pull/72 to be merged first ideally.

**Comment by johmsalas** [comment]
I'd like to fix first https://github.com/johmsalas/text-case.nvim/issues/61 and https://github.com/johmsalas/text-case.nvim/issues/62

**Comment by peterfication** [comment]
Sounds like a good idea to me!

File: tests/textcase/plugin/options/default_keymappings_false_spec.lua (hunk 1 above)
Comment (peterfication) [reply tests/textcase/plugin/options/default_keymappings_false_spec.lua]: Done already.

File: lua/textcase/plugin/presets.lua [hunk 3]
```diff
@@ -100,4 +64,51 @@ M.setup = function(opts)
...
+    api.to_upper_phrase_case,
+    api.to_lower_phrase_case,
+  })
+end
+
+M.setup = function(opts)
+  local prefix = opts and opts.prefix or "ga"
+
+  local default_keymappings_enabled = true
+  if opts and opts.default_keymappings_enabled ~= nil then
```
Comment (johmsalas) [inline lua/textcase/plugin/presets.lua:105]: There are now enough tests to verify the default key mapping does not break with this change

Thanks for your patience on the LSP and Telescope setups
And for adding n and v keybinding tests

**Review by johmsalas — APPROVED** [review]
Awesome work on defining the edge cases for the assertions 🥳

**Review by johmsalas — APPROVED** [review]
